File: problems/1131_maximum_of_absolute_value_expression/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int maxAbsValExpr(vector<int>& arr1, vector<int>& arr2) {
        int n = arr1.size(), result = 0;
        vector<pair<int,int>> dirs = {{1,1},{1,-1},{-1,1},{-1,-1}};
        for (auto [p, q] : dirs) {
            int minVal = INT_MAX, maxVal = INT_MIN;
            for (int i = 0; i < n; i++) {
                int val = p * arr1[i] + q * arr2[i] + i;
                minVal = min(minVal, val);
                maxVal = max(maxVal, val);
            }
            result = max(result, maxVal - minVal);
        }
        return result;
    }
};

File: problems/1131_maximum_of_absolute_value_expression/solution.py
class Solution:
    def maxAbsValExpr(self, arr1: list[int], arr2: list[int]) -> int:
        n = len(arr1)
        result = 0
        for p, q in [(1, 1), (1, -1), (-1, 1), (-1, -1)]:
            min_val = float('inf')
            max_val = float('-inf')
            for i in range(n):
                val = p * arr1[i] + q * arr2[i] + i
                min_val = min(min_val, val)
                max_val = max(max_val, val)
            result = max(result, max_val - min_val)
        return result
